### PR TITLE
Produce IAU compliant source names

### LINKED ISSUE #617: Generated source names violate IAU convention
The pipeline generates source names based on the coordinates, i.e. VAST JHHMMSS.s±DDMMSS. While converting the source coordinate to a string of the desired precision, it rounds rather than truncates. The IAU convention is to truncate coordinates. See <https://cdsweb.u-strasbg.fr/Dic/iau-spec.html#S3.2.1>.

Relevant code:

https://github.com/askap-vast/vast-pipeline/blob/1f57760ad7a426049e37fb014b183a02327ea003/vast_pipeline/pipeline/model_generator.py#L58-L62

https://github.com/askap-vast/vast-pipeline/blob/1f57760ad7a426049e37fb014b183a02327ea003/vast_pipeline/utils/utils.py#L116-L179

### PR DESCRIPTION
**Pull request by marxide**
Ensures that source names saved in the Source model are IAU compliant by truncating coordinates, instead of rounding. Includes a database migration to update the existing names in the database.

Fixes #617.

### CODE REVIEW

**Comment by marxide** [comment]
@ajstewart I think I've done this right, but if you can think of another coordinate I should add to the unit test please let me know. Maybe I've missed some edge case?

**Review by ajstewart — COMMENTED** [review]
I think all the test cases are covered but see my comment about the method. Not sure if I've gone through the process correctly but I have a feeling there's a very specific case the truncate might fail.

A couple of other things:
1. I suspect this code was somewhat inherited from TraP - I wondered if astropy has easy functionality to replace it? Or if there is a reason why we need to write this conversion ourselves I'm forgetting.
2. I think this will also mean that the parquet sources file will be out of sync for all the runs. I have a feeling the name column is included in that.

File: vast_pipeline/utils/utils.py [hunk 1]
```diff
@@ -134,20 +138,38 @@ def deg2dms(deg: float, dms_format: bool = False, precision: int = 2) -> str:
...
+    # if truncating, render the seconds with an extra 2 decimal places
+    # 2 dp is needed to avoid the formatter rounding e.g. 9.9 -> 10.0 when precision = 0
+    precision = precision + 2 if truncate else precision
     sec_width = 3 + precision if precision > 0 else 2
 
-    if dms_format:
-        return f'{signchar}{sex[0]:02d}d{sex[1]:02d}m{sex[2]:0{sec_width}.{precision}f}s'
+    degrees_str = f'{signchar}{degrees:02d}'
+    minutes_str = f'{minutes:02d}'
+    seconds_str = f'{seconds:0{sec_width}.{precision}f}'
```
[the remark this answers is not in the document]
Comment (marxide) [reply vast_pipeline/utils/utils.py]: Good catch, thanks.

> would it not be safer to completely ignore the string formatting for the truncate case and just split the string version of the seconds by `.` and take the first character?

I tried this approach but it didn't work well when the input coordinate precision is low. The deg2sex function can sometimes return a slightly off seconds value due to floating-point arithmetic (e.g. 75:32:28 returns 27.99999... seconds, so converting that to a string and splitting on the decimal point is going to end up with the wrong result).

**Comment by marxide** [comment]
> 1. I suspect this code was somewhat inherited from TraP - I wondered if astropy has easy functionality to replace it? Or if there is a reason why we need to write this conversion ourselves I'm forgetting.

I couldn't find one. There's a bit in the coordinates docs that gives an example of converting a coordinate to an IAU formatted string, but it rounds!

> 2. I think this will also mean that the parquet sources file will be out of sync for all the runs. I have a feeling the name column is included in that.

I don't see a name column in the sources parquet files.

File: vast_pipeline/utils/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajstewart) [reply vast_pipeline/utils/utils.py]: I think what you've changed it to looks good. I was going to say do we need that `deg2sex` function, the answer looks to be no!

**Comment by ajstewart** [comment]
> > 1. I suspect this code was somewhat inherited from TraP - I wondered if astropy has easy functionality to replace it? Or if there is a reason why we need to write this conversion ourselves I'm forgetting.
> 
> I couldn't find one. There's a bit in the coordinates docs that gives an example of converting a coordinate to an IAU formatted string, but it rounds!

Ha, great!

> > 2. I think this will also mean that the parquet sources file will be out of sync for all the runs. I have a feeling the name column is included in that.
> 
> I don't see a name column in the sources parquet files.

Oh yeah, my memory is off. I wonder why we didn't include that in the output? That would be useful I would have thought.

Odd test failures?

**Comment by marxide** [comment]
> Odd test failures?

Aggghhh. I have no idea what's causing these yet. Like last time, they seem unrelated to the PR. Probably another dependency problem? I'll dig deeper later.

**Comment by ajstewart** [comment]
> Aggghhh. I have no idea what's causing these yet. Like last time, they seem unrelated to the PR. Probably another dependency problem? I'll dig deeper later.

Having a quick look the error comes from handling the fits during the forced extraction. I'm wondering if there was a hiccup with the data download - the wget logs seem to suggest yes. Perhaps try running them again and they might pass.
